events/nats_config.go

```diff
@@ -0,0 +1,198 @@
+package events
+
+import (
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	// nats server connection timeout
+	connectTimeout = 100 * time.Millisecond
+
+	// reconnect jitter
+	reconnectJitter = 10 * time.Second
+
+	// reconnect wait
+	reconnectWait = 10 * time.Second
+
+	// subscription callback timeout
+	subscriptionCallbackTimeout = 5 * time.Second
+
+	// Nak message with delay
+	nakDelay = 5 * time.Minute
+
+	// consumer defaults
+	consumerAckWait       = 5 * time.Minute
+	consumerMaxAckPending = 100
+	consumerDeliverPolicy = nats.DeliverAllPolicy
+)
+
+// NatsOptions holds the configuration parameters to setup NATS Jetstream.
+type NatsOptions struct {
+	// URL is the NATS server URL
+	URL string `mapstructure:"url"`
+
+	// AppName is the name of the application connecting to the
+	// NATS stream, this parameter is used to open the NATS connection
+	// and bind as a durable consumer.
+	AppName string `mapstructure:"app_name"`
+
+	// NATS stream user, when no creds file is provided.
+	StreamUser string `mapstructure:"stream_user"`
```

Comment:
i would get rid of these alternative auth methods

Reply:
Yep, as mentioned in the previous comment, I'll remove these in a separate PR.